addons/l10n_id_efaktur/models/account_move.py

```diff
@@ -62,7 +62,7 @@ def _compute_need_kode_transaksi(self):
     def _constraint_kode_ppn(self):
         ppn_tag = self.env.ref('l10n_id.ppn_tag')
         for move in self.filtered(lambda m: m.l10n_id_kode_transaksi != '08'):
-            if any(ppn_tag.id in line.tax_tag_ids.ids for line in move.line_ids if line.exclude_from_invoice_tab is False) and any(ppn_tag.id not in line.tax_tag_ids.ids for line in move.line_ids if line.exclude_from_invoice_tab is False):
+            if any(ppn_tag.id in line.tax_tag_ids.ids for line in move.line_ids if line.exclude_from_invoice_tab is False and not line.display_type) and any(ppn_tag.id not in line.tax_tag_ids.ids for line in move.line_ids if line.exclude_from_invoice_tab is False and not line.display_type):
```

Comment:
I'd split this if to make more explicit; it's becoming unreadable.

Something with intermediate variables like:

contains_line_with_ppn = condition 1
contains_line_without_ppn = conditions 2
if contains_line_with_ppn and contains_line_without_ppn:
...

Also, writing 
not line.exclude_from_invoice_tab 
direcly would be cleaner than 
line.exclude_from_invoice_tab is False 

Let's make it more standard-looking :)